### PR TITLE
chore: upgrade to webpack@5.0.0

### PR DESCRIPTION
**Pull request by merceyz**
**What's the problem this PR addresses?**

We're a bit behind on the webpack version, while not a problem, the latest webpack version doesn't mangle the output as much as previous versions so it's easier to debug the build output when needed.

For example looking at the `.pnp.js` file, the output is much closer to what we actually write in the source
```diff
- this.realFs.copyFile(sources_path/* npath.fromPortablePath */.cS.fromPortablePath(sourceP), sources_path/* npath.fromPortablePath */.cS.fromPortablePath(destP), flags, this.makeCallback(resolve, reject));
+ this.realFs.copyFile(npath.fromPortablePath(sourceP), npath.fromPortablePath(destP), flags, this.makeCallback(resolve, reject));
```

The size is also reduced slightly, the CLI is reduced by 5024 bytes (1 869 733 to 1 864 709) and the source of the `.pnp.js` file by 7661 bytes (564 594 to 556 933)

**How did you fix it?**

Upgraded webpack

**Checklist**
- [x] I have read the [Contributing Guide](https://yarnpkg.com/advanced/contributing).
- [x] I have set the packages that need to be released for my changes to be effective.
- [x] I have verified that all automated PR checks pass.

### CODE REVIEW

File: packages/yarnpkg-builder/sources/tools/makeConfig.ts [hunk 1]
```diff
@@ -36,7 +36,7 @@ export const makeConfig = (config: webpack.Configuration): webpack.Configuration
   mode: `none`,
   devtool: false,
 
-  target: `node`,
+  target: `node10.19`,
```
Comment (merceyz) [inline packages/yarnpkg-builder/sources/tools/makeConfig.ts:39]: This made no difference to the build output, but it doesn't hurt either

File: packages/yarnpkg-pnp/package.json [hunk 2]
```diff
@@ -11,15 +11,14 @@
...
-    "webpack-cli": "^4.0.0-beta.8",
-    "webpack-sources": "^1.4.3"
+    "webpack": "5.0.0-beta.30",
+    "webpack-cli": "^4.0.0-beta.8"
   },
   "scripts": {
-    "build:pnp:hook": "run pnpify webpack-cli --config webpack.config.hook.js",
-    "build:pnp": "run pnpify webpack-cli --config webpack.config.pkg.js",
+    "build:pnp:hook": "run webpack-cli --config webpack.config.hook.js",
+    "build:pnp": "run webpack-cli --config webpack.config.pkg.js",
```
Comment (merceyz) [inline packages/yarnpkg-pnp/package.json:21]: `pnpify` isn't needed to run the build, unsure if it was here as a "test" case

File: packages/yarnpkg-pnp/webpack.config.hook.js [hunk 3]
```diff
@@ -1,5 +1,7 @@
+require(`@yarnpkg/monorepo/scripts/setup-ts-execution`);
```
Comment (merceyz) [inline packages/yarnpkg-pnp/webpack.config.hook.js:1]: `@yarnpkg/builder/sources/tools/makeConfig` is a ts file so need to setup `@babel/register`

**Comment by merceyz** [comment]
`webpack@5.0.0-beta.30` fails to handle the `this` here, making it undefined 
https://github.com/yarnpkg/berry/blob/e6e035c901ce921af890be9401076f0123e5393d/packages/yarnpkg-core/sources/Project.ts#L1371
https://github.com/yarnpkg/berry/pull/1832/checks?check_run_id=1105811689#step:5:26
`webpack@5.0.0-beta.29` handles it correctly.

To reproduce:
- Clone the repo
- `yarn install`
- `yarn build:cli --no-minify`
- search `packages/yarnpkg-cli/bundles/yarn.js` for `undefined.configuration`

cc @sokra

**Comment by sokra** [comment]
Interesting... I look into it

**Comment by merceyz** [comment]
@sokra Issue fixed :tada:, thank you! Though it seems a new one has been introduced (or I didn't notice before), the build output contains a lot of double semicolons now (`;;`)

File: .pnp.js [hunk 4]
```diff
@@ -46157,87 +46415,31 @@ run();
...
-module.exports = require("module");
-
-/***/ }),
-
-/***/ 87:
-/***/ ((module) => {
-
-"use strict";
-module.exports = require("os");
+module.exports = require("module");;
```
Comment (merceyz) [inline .pnp.js:46434]: Externals have an extra semicolon, but that shouldn't cause any issues, so it's not a blocker

cc @sokra

**Pull request merged**